File: mobile/lib/models/restaurant.dart
class Restaurant {
  final int id;
  final String name;
  final String description;
  final String? coverImageUrl;
  final double rating;
  final bool isOpen;

  Restaurant({
    required this.id,
    required this.name,
    required this.description,
    this.coverImageUrl,
    required this.rating,
    required this.isOpen,
  });

  factory Restaurant.fromJson(Map<String, dynamic> json) {
    return Restaurant(
      id: json['id'],
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      coverImageUrl: json['coverImageUrl'],
      rating: (json['rating'] ?? 0).toDouble(),
      isOpen: json['isOpen'] ?? true,
    );
  }
}

File: backend/src/main/java/com/foodgo/entity/Restaurant.java
package com.foodgo.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "restaurants")
@Getter
@Setter
public class Restaurant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "owner_id", nullable = false)
    private Long ownerId;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String address;

    @Column(name = "cover_image_url")
    private String coverImageUrl;

    private BigDecimal rating;

    @Column(name = "is_open")
    private Boolean isOpen = true;

    @Column(name = "created_at", insertable = false, updatable = false)
    private LocalDateTime createdAt;
}
